### PR TITLE
net: add `Ensure{any}Banman`

### PR DESCRIPTION
**Pull request by brunoerg**
This PR adds `Ensure{any}Banman` functions to avoid code repetition and make it cleaner. Same approach as done with argsman, chainman, connman and others.

### CODE REVIEW

**Comment by davidgumberg** [comment]
ACK [2d955ff](https://github.com/bitcoin/bitcoin/pull/27069/commits/2d955ff006b8949017fe617c35cfc1cfe117db6e)

File: src/rpc/net.cpp [hunk 1]
```diff
@@ -702,9 +702,7 @@ static RPCHelpMan setban()
         throw std::runtime_error(help.ToString());
     }
     NodeContext& node = EnsureAnyNodeContext(request.context);
```
Comment (MarcoFalke) [inline src/rpc/net.cpp:704]: why not use EnsureAnyBanman here?

Comment (brunoerg) [reply src/rpc/net.cpp:704]: Because we need `connman` to disconnect the node right after banning.

Comment (brunoerg) [reply src/rpc/net.cpp:704]: e.g:
```cpp
banman.Ban(subNet, banTime, absolute);
if (node.connman) {
    node.connman->DisconnectNode(subNet);
}
```

**Pull request merged**